### ISSUE TITLE
Recommendations for self-use

### ISSUE THREAD

**Issue by boognish-rising** [NullPounce/pounce-keys#11]
As the title suggests, I'm trying to determine if Pounce-Keys can be used as a personal keylogger. I go into (much) detail below but 

**TL;DR:**

I just don't know how to use Pounce-Keys with the target and listener as one in the same device. Any help in this regard would be greatly appreciated, however if personal use is out of the scope of this app, just say so and feel free to close the issue. Or I'll close it once I get clarification/confirmation on whether Pounce-Keys can fill the role of "personal keylogger."

**Long version:**

Keyloggers get a lot of shit, some of it may be justified but the program/code itself isn't malicious/nefarious. Regardless, the train left the station long ago where its PR is concerned. As such, there are very few personal keyloggers for Android. Even fewer that are active/maintained, functional _and_ open source. 

Inputting Plus: Ctrl + Z/F/C/V used to be perfect for my needs (though I have always wished that the 7 day log/history limit could be made indefinite. That and the ability to filter by app but aside from these 2 minor gripes, it was exactly what I needed it to be otherwise). Sadly, it was abandoned long ago - the last update was ≈4½ years ago. All despite over 100K DLs @ $2 a pop & positive press from Android Police, LifeHacker, etc. It's not as though its dev retired from developing Android apps. He's actively developing/maintaining other apps & just released a ChatGPT overlay app less than 3mo ago. I still use Inputting+, but it's buggy as hell and breaks more with every major OS update. I anticipate it will be completely useless soon. 

The only other similar app on the Play Store that comes close (lacks the helper overlay and date filter/calendar BUT has a per app filter & allows for indefinite log history, albeit with a performance hit after a couple few months), is an app called Type Keeper but A) it's also been abandoned, receiving is last update ≈4yrs ago, only a few months after its release. B) I don't think it's FLOSS (it also has ads with no option to remove them via donation or otherwise but they're unobtrusive). 

Needless to say, I need a new Android keylogger. As mentioned, this is strictly for personal use. No judgement reserved for anyone using Pounce-Keys in more of a surveillance capacity; no one, myself included, has any business casting aspersions upon another without walking in their shoes, which... ya know, can't be done. 

Regardless, I have no such use case myself, however, between app crashes, OS crashes and temperamental text fields that balk at the idea of preserving text long enough for the user to reference a recent app, only to return 30 seconds later to find the text field and all progress wiped, I **_do_** have a powerful need to access my own key logs. A need which is almost never satisfied by the system clipboard, which has very limited functionality and is if **zero** help if one hasn't copied the lost text. Crash resistant? Methinks not. 

I discovered Pounce-Keys while doing some background research for a post in some hacking/infosec/privacy sub or other and DL'd it without fully grasping the fact that I'm not likely an ideal user but in the process of learning as much, I found that in every other regard, it's perfect. I just don't know how to use it in a way where the target and listener are one in the same device (that's my TL;DR for this behemoth).

Any help in this regard would be greatly appreciated, however if my use case is truly out of the scope of this app, just say so and feel free to close the issue. Or I'll close it once I get clarification/confirmation on whether Pounce-Keys can fill the role of "personal keylogger."

Thanks & sorry for the novel! 

I'm long-windedAF

**Comment by NullPounce** [comment]
thank you for putting a smile on my face, have not been active much, yes I've used this during testing to copy my google auth code without even touching my phone. There are 3 versions, all versions you must install the apk then enable its service in accessibility services, also BEFORE installing you should edit it with https://qwertycube.com/apk-editor-studio/    "apk editor" or it wont work. read the release notes for each version. for example the Gmail version you'd need to google how to create you're google "app" password which is different than your login, you'd follow the release notes and open the apk in the editor then look for search, then search and replace ALL. the code and instructions have placeholders like "insert-email-or-pass-here" and you are to edit it.

gmail is on a buffer that you can edit, so it sends the logs in chunks to prevent spam "release notes say what to look for". so for testing just spam they keyboard for a couple mins to check for the email, this version just needs the email and app password with an optional buffer number to edit i believe.

now the   2.1 stealth update version requires java on windows or linux, i prefer to just open the listener in the exe version and just click listen, now this will work on the same subnet and or wifi and will only work if you open up the port you listen on, i've used linode with phone data for just 5 bucks a month and it's stable as long as you're not allowing every single ip. now this is a live version with a handfull of filter tabs for each app. just make sure the port you use isnt in use, make sure it's allowed in firewall, works perfect for local use.

discord version same thing, use apk edit then in your channel get the bot token and enable the service on the phone, this one is also live and may crash from possible spam, although i never seen it happen.

glad people have alternative uses for this, can also be used to test bank apps against this, example firefox url bar is protected but not the google search page itself.

I just installed java jdk 19 real quick in windows and scrolled to the bottom of the release, downloaded the windows and linux rar and just ran the jar file, clicked ok to firewall and whatnot, made sure the ip was right and it was good to go, this is the live version but i like the gmail on as you dont need port forward or a whole server or pc running

**Comment by rohittotlani** [comment]
Hey @boognish-rising 
How were you able to pounce-keys working?
As per the instructions, we need to enter the gmail id and app password after opening the apk in apk editor studio; but I am unable to find the fields to replace those values with.
Any help would be well regarded.

**Comment by boognish-rising** [comment]
@NullPounce  - sorry for the delayed response but THANK YOU for going into the level of detail that you did in your response! I would never have expected you to go to such lengths, as I try not to be a burden on indy/FLOSS devs who are already stretched thin as it is. Especially not when detailed documentation is readily available. I  skimmed said documentation but in my cursory viewing, came away with the impression alluded to in the  OP, i.e., that my use case was somewhat out of scope and as such, wanted to make sure that it _was_ in fact achievable before undertaking the not so trivial process necessary for arriving at a fully operational, running instance of Pounce-Keys. Assuming it's feature set satisfies the intended purpose, that is. After reading your response,  I think I may have been a tad hasty in reaching that conclusion, however, and while a bit more convoluted than it might otherwise be if it were conceived of/developed wiith the  "personal keylogger" use case in mind, thanks to the insight gleaned from your comprehensive response, I believe I see a path for satisfying my use case after-all  

PS - im really happy to hear this issue/general line of inquiry was well received and even made you smile! I'm always afraid that my long-winded nature will be viewed as extraneous detail or fluff that shouldn't have been included. Granted, I'm long-winded by nature and in most text I compose regardless of platform or what it's for, etc. reflects as much but especially when intercting with app devs, I would _think_ that the more detail - the better. At least when submitting any type of bug report/issue, etc That, or dismissed outright as the ravings of a lunatic lol

@rohittotlani - thanks for the question, as I actually noticed the notification it spurred. Alas, I can't (yet) answer/be of much help since as I just alluded to, I haven't yet actually put PK to the test to see if it'll work as hoped but being highly configurable, I believe such a use case can be accomodated. I'm going to close this issue, since it wasn't so much of an issue as it was a general question but once I get some tiime to put it thtough its paces, so to speak, I'll report back and fill you in/hopefully answer any questions. If you experience success sooner and have anything that you think might be helpful to contribute, please do share. I'm not sure if we have the ability to re-open one another's issues but if so, feel free to reopen if you think it prudent. If not, you should still be able to comment on it, anyway.
